scripts/fuzz_opt.py

```diff
@@ -361,23 +432,45 @@ class CompareVMs(TestCaseHandler):
     def __init__(self):
         super(CompareVMs, self).__init__()
 
-        def byn_run(wasm):
-            return run_bynterp(wasm, ['--fuzz-exec-before'])
+        class BinaryenInterpreter:
+            name = 'binaryen interpreter'
+
+            def run(self, wasm):
+                return run_bynterp(wasm, ['--fuzz-exec-before'])
 
-        def v8_run(wasm):
-            run([in_bin('wasm-opt'), wasm, '--emit-js-wrapper=' + wasm + '.js'] + FEATURE_OPTS)
-            return run_vm([shared.V8, wasm + '.js'] + shared.V8_OPTS + ['--', wasm])
+            def can_run(self, wasm):
+                return True
 
-        def yes():
-            return True
+            def can_compare_to_self(self):
+                return True
 
-        def if_legal_and_no_nans():
-            return LEGALIZE and not NANS
+            def can_compare_to_others(self):
+                return True
 
-        def if_no_nans():
-            return not NANS
+        class D8:
+            name = 'd8'
 
-        class Wasm2C(VM):
+            def run(self, wasm):
+                run([in_bin('wasm-opt'), wasm, '--emit-js-wrapper=' + wasm + '.js'] + FEATURE_OPTS)
+                return run_vm([shared.V8, wasm + '.js'] + shared.V8_OPTS + ['--', wasm])
+
+            def can_run(self, wasm):
+                # INITIAL_CONTENT is disallowed because some initial spec testcases
+                # have names that require mangling, see
+                # https://github.com/WebAssembly/binaryen/pull/3216
```

Comment:
The specific problem is that an export with "-" in it is not a valid JS name. It can be used with quotes, but then there are other issues, like not working in an ES6 module import, it turns out (see link).